code/modules/projectiles/guns/energy/poweredcrossbow.dm

```diff
@@ -0,0 +1,90 @@
+/obj/item/gun/energy/poweredcrossbow
+	name = "Powered Crossbow"
+	desc = "An handmade crossbow linked to an attached power cell, a weapon of desperation and ingenuity."
+	icon = 'icons/obj/guns/energy/poweredcrossbow.dmi'
+	icon_state = "crossbow"
+	item_state = "crossbow"
+	w_class = ITEM_SIZE_NORMAL
+	force = WEAPON_FORCE_PAINFUL
+	flags = CONDUCT
+	slot_flags = SLOT_BACK
+	wielded_item_state = "_doble"
+	charge_cost = 100
+	suitable_cell = /obj/item/cell/large
+	cell_type = /obj/item/cell/large
+	projectile_type = /obj/item/projectile/bullet/bolt
+	fire_delay = 12 //Equivalent to a pump then fire time
+	fire_sound = 'sound/weapons/tablehit1.ogg'
+	init_firemodes = list(
+		list(mode_name="Bolt", mode_desc="Fires a charged quarrel", projectile_type=/obj/item/projectile/bullet/bolt, charge_cost=100, icon="kill"),)
+	price_tag = 200
+	spawn_tags = SPAWN_TAG_GUN_HANDMADE
+	matter = list(MATERIAL_STEEL = 3, MATERIAL_PLASTIC = 5, MATERIAL_WOOD = 5)
+	twohanded = FALSE
+	var/consume_cell = TRUE
+	var/obj/item/ammo_casing/crossbow/bolt/bolt
+	var/is_drawn
+	init_recoil = RIFLE_RECOIL(1)
+	safety = FALSE
+	restrict_safety = TRUE
+
+/obj/item/gun/energy/poweredcrossbow/consume_next_projectile()
+	if(!cell) return null
+	if(!ispath(projectile_type)) return null
+	if(consume_cell && !cell.checked_use(charge_cost))
+		visible_message(SPAN_WARNING("\The [cell] of \the [src] burns out!"))
+		qdel(cell)
+		cell = null
+		playsound(loc, 'sound/weapons/Egloves.ogg', 50, 1, -1)
+		new /obj/effect/decal/cleanable/ash(get_turf(src))
+		return new projectile_type(src)
```

Comment:
You should change the type of projectile_type otherwise it will shoot a bolt instead of the empty battery after loosening the safety bolts.